File: Julia/mtx_loop.jl
# Undergraduate Student: Arturo Burgos
# Professor: João Rodrigo Andrade
# Federal University of Uberlândia - UFU, Fluid Mechanics Laboratory - MFLab, Block 5P, Uberlândia, MG, Brazil

using Printf # Just the same as import a module
using Statistics
using CPUTime

println("Choose a N value: ")
N = chomp(readline()) # Here I get the char from the user
n = parse(Int32,N)  # Here I convert the char into an integer(32bits)

A = rand(1:20000,n,n,3);

println("\n\nThe A[1,1,1] is: ",A[1,1,1])
println("\nThe A[1,1,2] is: ",A[1,1,2])
println("\nThe A[1,1,3] is: ",A[1,1,3])
println("\n\nThe A[n,n,1] is: ",A[n,n,1])
println("The A[n,n,2] is: ",A[n,n,2])
println("The A[n,n,3] is: ",A[n,n,3])

function loop(mtx)

    for i = 1:n
        for j=1:n
            mtx[i, j, 1] = mtx[i, j, 2]
            mtx[i, j, 3] = mtx[i, j, 1]
        end
    end
    sleep(1)
    return mtx # retornar algo à variavel alocada, no caso "A"
end

A = loop(A);

println("\n\nThe new A[1,1,1] is: ",A[1,1,1])
println("\nThe new A[1,1,2] is: ",A[1,1,2])
println("\nThe new A[1,1,3] is: ",A[1,1,3])
println("\n\nThe A[n,n,1] is: ",A[n,n,1])
println("The A[n,n,2] is: ",A[n,n,2])
println("The A[n,n,3] is: ",A[n,n,3])

@time @CPUtime loop(A)

File: Fortran/1_ex/mtx_loop.f90
program mtx_loop

    !Undergraduate Student: Arturo Burgos
    !Professor: João Rodrigo Andrade
    !Federal University of Uberlândia - UFU, Fluid Mechanics Laboratory
    !MFLab, Block 5P, Uberlândia, MG, Brazil

    !This program will read the data in the txt file

    implicit none !To avoid variable definition problems, checking if they are correctly defined
    integer :: i, j
    real, DIMENSION(5000,5000,3) :: mtx
    DOUBLE PRECISION :: start, finish

    call random_number(mtx)

    write(*,'(A/)') 
    print *,'the mtx(1,1,1) now is: ', mtx(1,1,1)
    print *,'the mtx(1,1,2) now is: ', mtx(1,1,2)
    print *,'the mtx(1,1,3) now is: ', mtx(1,1,3)
    print *,'the mtx(n,n,1) now is: ', mtx(5000,5000,1)
    print *,'the mtx(n,n,2) now is: ', mtx(5000,5000,2)
    print *,'the mtx(n,n,3) now is: ', mtx(5000,5000,3)
    write(*,'(//A/)')

    call cpu_time(start)

    do i = 1,5000

        do j =1,5000
            
            mtx(i, j, 1) = mtx(i, j, 2)
            mtx(i, j, 3) = mtx(i, j, 1)

        end do

    end do

    call cpu_time(finish)

    write(*,'(A/)') 
    print *,'the mtx(1,1,1) now is: ', mtx(1,1,1)
    print *,'the mtx(1,1,2) now is: ', mtx(1,1,2)
    print *,'the mtx(1,1,3) now is: ', mtx(1,1,3)
    print *,'the mtx(n,n,1) now is: ', mtx(5000,5000,1)
    print *,'the mtx(n,n,2) now is: ', mtx(5000,5000,2)
    print *,'the mtx(n,n,3) now is: ', mtx(5000,5000,3)
    write(*,'(A/)') 
    print '("Time of operation is: ",f6.5," seconds")', finish - start


end program mtx_loop